### PR TITLE
Update min sdk version and remove redundant checks

### LINKED ISSUE #1131: Can we update minSdkVersion?
Google's [version dashboard](https://developer.android.com/about/dashboards/index.html) is reporting very small numbers for certain versions of Android.  Particularly,

| API Level | Distribution |
| ---- | ----- |
| API 15 (Ice Cream Sandwich) | **0.4 %** |
| API 16 (Jelly Bean) | **1.7%** |
| API 17 (Jelly Bean) | **2.6%** |
| API 18 (Jelly Bean) | **0.7%** |

Does the Google Play Store give insight to our actual users and the versions in use?

In the absence of hard numbers from actual users, could we move up to API 19 (KitKat) or at the very least, API 16?

### PR DESCRIPTION
**Pull request by maskaravivek**
**Description (required)**

Fixes #1131 

What changes did you make and why?

- updated minSdk version
- removed redundant checks for lower SDK versions

**Tests performed (required)**

built and ran the `betaDebug` variant of the app. It seems to be working fine.

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/utils/DialogUtil.java [hunk 1]
```diff
@@ -55,10 +53,7 @@ public static void showSafely(Activity activity, Dialog dialog) {
             return;
         }
 
-        boolean isActivityDestroyed = false;
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1) {
-            isActivityDestroyed = activity.isDestroyed();
-        }
+        boolean isActivityDestroyed = activity.isDestroyed();
```
Comment (domdomegg) [inline app/src/main/java/fr/free/nrw/commons/utils/DialogUtil.java]: This variable probably not necessary after refactor

File: app/src/main/java/fr/free/nrw/commons/utils/DialogUtil.java [hunk 2]
```diff
@@ -30,9 +30,7 @@ public static void dismissSafely(@Nullable Activity activity, @Nullable DialogFr
             return;
         }
 
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1) {
-            isActivityDestroyed = activity.isDestroyed();
-        }
+        isActivityDestroyed = activity.isDestroyed();
```
Comment (domdomegg) [inline app/src/main/java/fr/free/nrw/commons/utils/DialogUtil.java]: This variable probably not necessary after refactor

File: app/src/main/java/fr/free/nrw/commons/utils/DialogUtil.java [hunk 3]
```diff
@@ -83,9 +78,7 @@ public static void showSafely(FragmentActivity activity, DialogFragment dialog)
...
```
Comment (domdomegg) [inline app/src/main/java/fr/free/nrw/commons/utils/DialogUtil.java]: This variable probably not necessary after refactor

File: app/src/main/java/fr/free/nrw/commons/utils/DialogUtil.java (hunk 2 above)
Comment (maskaravivek) [reply app/src/main/java/fr/free/nrw/commons/utils/DialogUtil.java]: @domdomegg This function is not getting used anywhere in the codebase. Is it fine if I remove the whole function altogether? Or should that be done in another PR?

File: app/src/main/java/fr/free/nrw/commons/utils/DialogUtil.java (hunk 1 above)
Comment (domdomegg) [reply app/src/main/java/fr/free/nrw/commons/utils/DialogUtil.java]: Yeah - let's do it in another PR

**Pull request merged**